### ISSUE TITLE
[legal] do you have the licenses of all the resources icons ?

### ISSUE THREAD

**Issue by picca** [silx-kit/silx#1158]
Everythings in the title :))

I try to use cme in order to generate and upgrade the copyright file. But I do not have the license of all theses svg, png files.

**Comment by picca** [comment]
When I run

`cme upgrade dpkg-copyright`

I get this

```The following paths are missing information:
- doc/source/conf.py: missing copyright and license
- doc/source/img/silx.ico: missing copyright and license
- silx/gui/plot/MPLColormap.py: missing copyright and license
- silx/image/bilinear.c: missing copyright and license
- silx/image/shapes.c: missing copyright and license
- silx/image/sift.py: missing copyright and license
- silx/io/specfile/specfile.c: missing copyright and license
- silx/math/combo/combo.c: missing copyright and license
- silx/math/fit/filters/filters.c: missing copyright and license
- silx/math/fit/functions/functions.c: missing copyright and license
- silx/math/fit/peaks/peaks.c: missing copyright and license
- silx/math/histogramnd/chistogramnd.c: missing copyright and license
- silx/math/histogramnd/chistogramnd_lut.c: missing copyright and license
- silx/math/marchingcubes/marchingcubes.cpp: missing copyright and license
- silx/math/medianfilter/medianfilter.cpp: missing copyright and license
- silx/opencl/utils.py: missing copyright
You may want to add a line in debian/fill.copyright.blanks.yml
```

**Comment by picca** [comment]
I think that some of the problem are due to licensecheck wwhcih does not regognize the license header.
Since it work with a bunch of files, it would be  nice to investigaet and understand why it fail sometimes.

**Comment by vasole** [comment]
- The icons were generated at the ESRF so, they are under the same license as silx (MIT).

- Concerning the c files, some (all?) of them are generated by cython.

- The file silx/opencl/utils.py contains the license but not the explicit Copyright leading sentence.

**Comment by PiRK** [comment]
- `silx/gui/plot/MPLColormap.py` has been recently moved to silx/gui/plot/matplotlib/Colormap.py`. The copyright header seems to not be standard (CC0). We could put our own copyrigth header, as far as I understand CC0 allows us to do whatever we want. But CC0 is more permissive than MIT, so I don't see the point of doing this.
 - `doc/source/conf.py` was auto-generated by sphinx, then slightly modified by us. Are we allowed to add our copyright header?
 - `silx/image/sift.py` is a one liner, but I will add our copyright anyway
 - `silx/opencl/utils.py` is not our standard header because the file was imported from another project, I will correct it.

All the `.c` and `.cpp` files in your `cme` log are cython generated files.

**Comment by PiRK** [comment]
I see that all the individual colormap files under CC0 license are properly listed in our copyright file, so I will just add `Colormap.py`

I don't understand the issue with icons. Shouldn't the following block encompass them?

```
Files: *
Copyright: 2004-2017 European Synchrotron Radiation Facility
           Data analysis unit (silx@esrf.fr) 
License: MIT
```

**Comment by t20100** [comment]
@PiRK:
- To me you can put the silx license to sphinx's `conf.py` file.
- You don't have to change `Colormap.py` copyright, it's a file of the project, it was not copied from elsewhere (`MPLColormap.py` was).

**Comment by t20100** [comment]
Icon files do not embed a copyright notice

**Comment by picca** [comment]
Hello, thanks for the information, when I look at the .svg files with inkscape. the license is "propriétary".
This is why I asked on this bug report. This is for documentation purpose.

I must convince the ftp-master team that the license of all files are DFSG compatible.
This is a boring work but essential.

This is also why I try to automate as much as possible the generation of the copyright file for Debian and Derivatives.
This require a few changes in some files in order for licensecheck to identify properly all the licences.


thanks for your help :)

**Comment by vallsv** [comment]
If you found license filed in SVG files, it is an issue. This icons are directly used, and are supposed to be as light as possible. Then AFAIK we clean up non-useful informations.

**Comment by picca** [comment]
This is not the all license text.

this is just the legal information when using inkscape.
It is possible to select a bunch of licenses via a checkbox.
In the file menu Ctrl+Shidt+D

**Comment by t20100** [comment]
We use scour (https://github.com/scour-project/scour) to optimize the SVG files, so unnecessary fields are removed, but maybe a few icons when through...

**Comment by picca** [comment]
Nice tool :), I discovered that the debian pacakge provide this

```
 This package also provides a dh_scour debhelper extension which optimizes all
 shipped SVGs during package build. If python-rsvg and python-cairo are
 available, it will also do a before/after comparison and discard the optimized
 image if they differ by more than 0.05%.
```

To my opinion, the license is wrong from the begining in the svg file. Did someone setup a MIT license in the Document properties ?

Cheers

**Comment by picca** [comment]
Looking at the scour documentation, I can read this.

```
Description-en: SVG scrubber and optimizer (Python 2 module)
 Scour is a Python module that aggressively cleans SVG files, removing a lot of
 unnecessary information that certain tools or authors embed into their
 documents.  The goal of scour is to provide an identically rendered image
 (i.e. a scoured document should have no discernable visible differences from
 the original file) while minimizing the file size.
 .
 WARNING: Scour is intended to be run on files that have been edited in Vector
 Graphics editors such as Inkscape or Adobe Illustrator.  Scour attempts to
 optimize the file, and as result, it will change the file's structure and
 (possibly) its semantics.  If you have hand-edited your SVG files, you will
 probably not be happy with the output of Scour.
 .
 Never use scour to overwrite your original file!
```

The git repository contain the original svg images ?

**Comment by vallsv** [comment]
No, the license is not wrong, just the header field is not defined. Inkscape is maybe defaulting it as "Copyrighted", which is not exact, cause nothing is defined, that's then not the right tool to check that. It's easy to check SVG files manually, that's just XML.

Do you have a list of faulty SVG files?

**Comment by vallsv** [comment]
We store on our Git repository the only version of the images we have. We dont want to have Inkscape format, we have pure SVG format which only contains rendering tags.

**Comment by t20100** [comment]
We want to store files converted with scour, but I checked, there is quite a few files which still contain some Inkscape fields...

**Comment by t20100** [comment]
And `scour` does not remove the license/copyright fields by default (but there is a `--remove-metadata` option for it).

@picca, is it OK if we remove the license metadata from each SVG file and say in the copyright file that it is MIT and copyrighted ESRF?

**Comment by vasole** [comment]
Can this be considered closed by #1173 ?

**Comment by picca** [comment]
Since silx was accepted, I guess :).
